spec/models/date_range_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+RSpec.describe DateRange do
+
+  describe '#dates' do
+      let(:conference) { FactoryGirl.build_stubbed(:conference)}
+    
+    it 'creates a date range for calling Models' do
+      expect(conference.dates).to be_an_instance_of DateRange
+    end
+  end
+  
+  describe '#compact' do
+    let (:conference)  { FactoryGirl.build_stubbed(:conference) }
+    subject { described_class.new(start_date: Date.parse('2015-12-31'), end_date: Date.parse('2015-12-31')) }
```

Comment:
Just a matter of style, but I think the tests are more meaningful if the `subject` reflects the actual "subject" of all expectations. So in order to dry this up a bit and make it more expressive I'd suggest to switch it to be actually `compact`:

```ruby
let(:start_date) { Date.parse('2015-12-31') }
let(:end_date)   { Date.parse('2015-12-31') }

subject { described_class.new(start_date: start_date, end_date: end_date).compact }

# etc.
```

And then, for different contexts, all you have to do is change the values for `starts_at` and `ends_at`:
```ruby
context 'when multiple dates in same month' do
  let(:start_date) { Date.parse('2015-12-30') }
  let(:end_date    { Date.parse('2015-12-31') }

  it 'returns compact days' do
    expect(subject).to eq ('30 - 31 Dec 2015')
  end
end
```

What do you think?